### PR TITLE
Add security context to stack

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes Add security context to a stack manifest

## Proposed changes
- Adds `security_context` as part of service specification
- Add tests for testing unmarshalling

### CODE REVIEW

File: pkg/model/stack_serializer.go [hunk 1]
```diff
@@ -692,6 +696,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+			if len(parts) != 2 {
+				return fmt.Errorf("SecurityContext is malformed. Only 'dddd:dddd' is supported")
+			}
+			runAsUser, err := strconv.Atoi(parts[0])
+			if err != nil {
+				return fmt.Errorf("Can not convert '%s' to a user", rawSecurityContext)
+			}
+			runAsGroup, err := strconv.Atoi(parts[1])
+			if err != nil {
+				return fmt.Errorf("Can not convert '%s' to a user", rawSecurityContext)
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:714]:
```suggestion
				return fmt.Errorf("Can not obtain GID from '%s'", parts[1])
```

File: pkg/model/stack_serializer.go [hunk 2]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+	err := unmarshal(&rawSecurityContext)
+	if err == nil {
+		if strings.Contains(rawSecurityContext, ":") {
+			parts := strings.Split(rawSecurityContext, ":")
+			if len(parts) != 2 {
+				return fmt.Errorf("SecurityContext is malformed. Only 'dddd:dddd' is supported")
+			}
+			runAsUser, err := strconv.Atoi(parts[0])
+			if err != nil {
+				return fmt.Errorf("Can not convert '%s' to a user", rawSecurityContext)
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:710]:
```suggestion
				return fmt.Errorf("Can not obtain UID from '%s'", parts[0])
```

File: pkg/model/stack_serializer.go [hunk 3]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+			}
+			runAsGroup, err := strconv.Atoi(parts[1])
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:712]:
```suggestion
			runAsGroup, err := strconv.ParseInt(parts[1], 10, 64)
```

File: pkg/model/stack_serializer.go [hunk 4]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+func (sc *StackSecurityContext) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var rawSecurityContext string
+	err := unmarshal(&rawSecurityContext)
+	if err == nil {
+		if strings.Contains(rawSecurityContext, ":") {
+			parts := strings.Split(rawSecurityContext, ":")
+			if len(parts) != 2 {
+				return fmt.Errorf("SecurityContext is malformed. Only 'dddd:dddd' is supported")
+			}
+			runAsUser, err := strconv.Atoi(parts[0])
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:708]:
```suggestion
			runAsUser, err := strconv.ParseInt(parts[0], 10, 64)
```

File: pkg/model/stack_serializer.go [hunk 5]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+			if err != nil {
+				return fmt.Errorf("Can not convert '%s' to a user", rawSecurityContext)
+			}
+			runAsGroup, err := strconv.Atoi(parts[1])
+			if err != nil {
+				return fmt.Errorf("Can not convert '%s' to a user", rawSecurityContext)
+			}
+			sc.RunAsUser = pointer.Int64(int64(runAsUser))
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:716]:
```suggestion
			sc.RunAsUser = &runAsUser
```

File: pkg/model/stack_serializer.go [hunk 6]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+			sc.RunAsGroup = pointer.Int64(int64(runAsGroup))
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:717]:
```suggestion
			sc.RunAsGroup = &runAsGroup
```

File: pkg/model/stack_serializer.go [hunk 7]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+		} else {
+			runAsUser, err := strconv.Atoi(rawSecurityContext)
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:719]:
```suggestion
			runAsUser, err := strconv.ParseInt(rawSecurityContext)
```

File: pkg/model/stack_serializer.go [hunk 8]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+			if err != nil {
+				return fmt.Errorf("Can not convert '%s' to a user", rawSecurityContext)
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:721]:
```suggestion
				return fmt.Errorf("Can not obtain UID from '%s'", rawSecurityContext)
```

File: pkg/model/stack_serializer.go [hunk 9]
```diff
@@ -692,6 +695,46 @@ func (warning *WarningType) UnmarshalYAML(unmarshal func(interface{}) error) err
...
+			}
+			sc.RunAsUser = pointer.Int64(int64(runAsUser))
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:723]:
```suggestion
			sc.RunAsUser = &runAsUser
```

File: pkg/model/stack_serializer.go [hunk 10]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/kballard/go-shellquote"
 	apiv1 "k8s.io/api/core/v1"
 	resource "k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/pointer"
```
Comment (jpf-okteto) [inline pkg/model/stack_serializer.go:26]:
```suggestion
```

**Review by pchico83 — APPROVED** [review]

**Review by jpf-okteto — CHANGES_REQUESTED** [review]
- use `strconv.ParseInt(s, 10, 64)` to obtain `int64` directly.
- use `&` instead of `pointer.Int64(i)`
- specific error messages on `dddd:dddd` format, user from `part[0]`, group from `part[1]`

**Review by jpf-okteto — APPROVED** [review]

**Pull request merged**